tests/unit/map_test.c

```diff
@@ -516,6 +516,154 @@ static void test_hashmap_degenerate_hash_fn(void)
     HashMapDestroy(hashmap);
 }
 
+#define ARRAY_STRING_KEY(last_char) {'k', 'e', 'y', last_char, '\0'}
+#define ARRAY_STRING_VALUE(last_char) {'v', 'a', 'l', 'u', 'e', last_char, '\0'}
+
+/**
+ * @brief Return pointer to heap allocated string constructed as
+ *        "keyX" where X is last_char.
+ */
+static char *MallocStringKey(char last_char)
+{
+    return xstrdup((char[]) ARRAY_STRING_KEY(last_char));
+}
+
+/**
+ * @brief Return pointer to heap allocated string constructed as
+ *        "valueX" where X is last_char.
+ */
+static char *MallocStringValue(char last_char)
+{
+    return xstrdup((char[]) ARRAY_STRING_VALUE(last_char));
+}
+
+static void test_array_map_insert(void)
+{
+    ArrayMap *m = ArrayMapNew(StringEqual_untyped, free, free);
+
+    // Test simple insertion
+    for (int i = 0; i < 3; ++i)
+    {
+        char* k = MallocStringKey('A' + i);
+        char* v = MallocStringValue('A' + i);
+        assert_int_equal(ArrayMapInsert(m, k, v), 2);
+    }  // size == 3
+
+    // Test replacing
+    for (int i = 0; i < 3; ++i)
+    {
+        char* k = MallocStringKey('A' + i);
+        char* v = MallocStringKey('0' + i);
+        assert_int_equal(ArrayMapInsert(m, k, v), 1);
+    }  // size == 3
+
+    // Fill array up to size 14
+    for (int i = 0; i < 14 - 3; ++i)
+    {
+        ArrayMapInsert(m, MallocStringKey('a' + i), MallocStringValue('a' + i));
+    }
+
+    // Test (no) insertion for size > 14
+    for (int i = 0; i < 3; ++i)
+    {
+        char* k = MallocStringKey('\0');
+        char* v = MallocStringValue('\0');
+        assert_int_equal(ArrayMapInsert(m, k, v), 0);
+        free(k);
+        free(v);
+    }
+
+    ArrayMapDestroy(m);
+}
+
+void test_array_map_get(void)
+{
+    ArrayMap *m = ArrayMapNew(StringEqual_untyped, free, free);
+
+    // Fill with some test strings
+    for (int i = 0; i < 5; ++i)
+    {
+        char *k = MallocStringKey('A' + i);
+        char *v = MallocStringValue('A' + i);
+        assert_int_equal(ArrayMapInsert(m, k, v), 2);
+    }
+
+    // Get all the test strings
+    for (int i = 0; i < 5; ++i)
+    {
+        char k[] = ARRAY_STRING_KEY('A' + i);
+        MapKeyValue *pair = ArrayMapGet(m, k);
+        assert_true(pair != NULL);
+        assert_string_equal(pair->key, (char[]) ARRAY_STRING_KEY('A' + i));
+        assert_string_equal(pair->value, (char[]) ARRAY_STRING_VALUE('A' + i));
+    }
+
+    // Get non existent keys
+    for (int i = 0; i < 5; ++i)
+    {
+        char k[] = ARRAY_STRING_KEY('\0');
+        MapKeyValue *pair = ArrayMapGet(m, k);
+        assert_true(pair == NULL);
+    }
```

Comment:
Get the same key 5 times?

Reply:
This is a refactoring error. I wanted to search 5 non existent keys so something like
`ARRAY_STRING_KEY('a' + i);` since 'a' is after 'A' so it wont be in the map